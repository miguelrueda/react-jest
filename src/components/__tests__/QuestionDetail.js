import { mapStateToProps, QuestionDetailDisplay } from "../QuestionDetail";
import renderer from "react-test-renderer";
import React from "react";

describe(`The Question Detail Component`, () => {

    describe(`The container element`, () => {
        describe(`mapStateToProps`, () => {
            it(`should map the state to props correctly`, () => {
                const sampleQuestion = {
                    question_id: 42,
                    body: "Space is big"
                };
                const appSate = {
                    questions: [
                        sampleQuestion
                    ]
                };
                const ownProps = {
                    question_id: 42,
                };
                const componentSate = mapStateToProps(appSate, ownProps);
                console.log(componentSate);

                expect(componentSate).toEqual(sampleQuestion);
            });
        });
    });

    describe(`The display element`, () => {
        it(`Should not regress`, () => {
            const tree = renderer.create(
                <QuestionDetailDisplay 
                    title="The meaning of life 2"
                    body="42"
                    answer_count={0}
                    tags={['hitchhiking']}
                />);
            expect(tree.toJSON).toMatchSnapshot();
        });
    });
});
