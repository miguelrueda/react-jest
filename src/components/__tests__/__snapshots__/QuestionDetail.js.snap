// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Question Detail Component The display element Should not regress 1`] = `[Function]`;
